store/src/main/java/org/apache/rocketmq/store/ConsumeQueue.java

```diff
@@ -833,7 +833,13 @@ private boolean putMessagePositionInfo(final long offset, final int size, final
                 }
             }
             this.setMaxPhysicOffset(offset + size);
-            return mappedFile.appendMessage(this.byteBufferIndex.array());
+            boolean appendResult;
+            if (messageStore.getMessageStoreConfig().isPutConsumeQueueDataByFileChannel()) {
+                appendResult = mappedFile.appendMessageUsingFileChannel(this.byteBufferIndex.array());
```

Comment:
`byteBufferIndex`, of which type is `ByteBuffer`, is transformed to `byte[]` and then transformed back into `ByteBuffer` in `DefaultMappedFile.appendMessageUsingFileChannel()`. IMO it's unnecessary.